src/library/autodj/autodjfeature.cpp

```diff
@@ -308,8 +310,22 @@ void AutoDJFeature::slotAddRandomTrack(bool) {
         retriveAttempts += 1;
     }
 
-    if (retriveAttempts == kMaxRetiveAttempts)
-        qDebug () << "Could Not Load Random Tracks";
+    if (retriveAttempts == kMaxRetiveAttempts || iTrackId == -1) {
+        qDebug () << "Could Not Load Random Tracks From Crate."
+    	        " Attempting to load from library";
+    	iTrackId = m_autoDjCratesDao.getRandomTrackIdFromLibrary();
```

Comment:
You have implemented a backup path for the case when m_autoDjCratesDao.getRandomTrackId() fails. Why not ..
This will work for me as well and reduced the complexity a lot.